Samba build system overrides the system-wide path to XML catalog files. If they happen
to live outside /etc/xml/catalog, the build is likely to fail when external entities
are processed.

This patch takes into account the $XML_CATALOG_FILES environment variable so that
the build succeeds on said platforms.

Signed-off-by: Lucas C. Villa Real <lucasvr@gobolinux.org>

diff -urp samba-4.6.0.orig/release-scripts/build-docs samba-4.6.0/release-scripts/build-docs
--- samba-4.6.0.orig/release-scripts/build-docs	2017-01-11 05:55:15.000000000 -0200
+++ samba-4.6.0/release-scripts/build-docs	2017-03-15 14:32:42.487351095 -0300
@@ -11,7 +11,9 @@ DOCSRCDIR=`dirname $0`/../docs-xml
 pushd $DOCSRCDIR || exit 1
 
 git clean -d -x -f
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+SYSTEM_XML_CATALOG_FILES_LIST=($(echo $XML_CATALOG_FILES))
+SYSTEM_XML_CATALOG_FILES=$(for i in ${SYSTEM_XML_CATALOG_FILES_LIST[@]}; do echo -n "file://$i "; done)
+export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml $SYSTEM_XML_CATALOG_FILES"
 autoconf && \
 	./configure --with-papersize=letter && \
 	make smbdotconf/parameters.all.xml && \
diff -urp samba-4.6.0.orig/release-scripts/build-htmlman-git samba-4.6.0/release-scripts/build-htmlman-git
--- samba-4.6.0.orig/release-scripts/build-htmlman-git	2017-01-11 05:55:15.000000000 -0200
+++ samba-4.6.0/release-scripts/build-htmlman-git	2017-03-15 14:34:05.593352654 -0300
@@ -34,7 +34,9 @@ autoconf || _exit "Error running autocon
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+SYSTEM_XML_CATALOG_FILES_LIST=($(echo $XML_CATALOG_FILES))
+SYSTEM_XML_CATALOG_FILES=$(for i in ${SYSTEM_XML_CATALOG_FILES_LIST[@]}; do echo -n "file://$i "; done)
+export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml $SYSTEM_XML_CATALOG_FILES"
 
 make htmlman || _exit "Error running make htmlman"
 
diff -urp samba-4.6.0.orig/release-scripts/build-htmlman-nogit samba-4.6.0/release-scripts/build-htmlman-nogit
--- samba-4.6.0.orig/release-scripts/build-htmlman-nogit	2017-01-11 05:55:15.000000000 -0200
+++ samba-4.6.0/release-scripts/build-htmlman-nogit	2017-03-15 14:34:14.376352819 -0300
@@ -32,7 +32,9 @@ autoconf || _exit "Error running autocon
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+SYSTEM_XML_CATALOG_FILES_LIST=($(echo $XML_CATALOG_FILES))
+SYSTEM_XML_CATALOG_FILES=$(for i in ${SYSTEM_XML_CATALOG_FILES_LIST[@]}; do echo -n "file://$i "; done)
+export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml $SYSTEM_XML_CATALOG_FILES"
 
 make clean || _exit "Error running make clean"
 
diff -urp samba-4.6.0.orig/release-scripts/build-manpages-git samba-4.6.0/release-scripts/build-manpages-git
--- samba-4.6.0.orig/release-scripts/build-manpages-git	2017-01-11 05:55:15.000000000 -0200
+++ samba-4.6.0/release-scripts/build-manpages-git	2017-03-15 14:34:22.120352964 -0300
@@ -34,7 +34,9 @@ autoconf || _exit "Error running autocon
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+SYSTEM_XML_CATALOG_FILES_LIST=($(echo $XML_CATALOG_FILES))
+SYSTEM_XML_CATALOG_FILES=$(for i in ${SYSTEM_XML_CATALOG_FILES_LIST[@]}; do echo -n "file://$i "; done)
+export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml $SYSTEM_XML_CATALOG_FILES"
 
 make manpages || _exit "Error running make manpages"
 
diff -urp samba-4.6.0.orig/release-scripts/build-manpages-nogit samba-4.6.0/release-scripts/build-manpages-nogit
--- samba-4.6.0.orig/release-scripts/build-manpages-nogit	2017-01-11 05:55:15.000000000 -0200
+++ samba-4.6.0/release-scripts/build-manpages-nogit	2017-03-15 14:34:28.513353084 -0300
@@ -32,7 +32,9 @@ autoconf || _exit "Error running autocon
 
 ./configure || _exit "Error running configure."
 
-export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml"
+SYSTEM_XML_CATALOG_FILES_LIST=($(echo $XML_CATALOG_FILES))
+SYSTEM_XML_CATALOG_FILES=$(for i in ${SYSTEM_XML_CATALOG_FILES_LIST[@]}; do echo -n "file://$i "; done)
+export XML_CATALOG_FILES="file:///etc/xml/catalog file://$(pwd)/build/catalog.xml $SYSTEM_XML_CATALOG_FILES"
 
 make clean || _exit "Error running make clean"
 
--- samba-4.6.0/buildtools/wafsamba/wafsamba.py.orig	2017-03-15 14:45:49.351365857 -0300
+++ samba-4.6.0/buildtools/wafsamba/wafsamba.py	2017-03-15 14:51:05.980371797 -0300
@@ -907,10 +907,13 @@ Build.BuildContext.MANPAGES = MANPAGES
 
 def SAMBAMANPAGES(bld, manpages, extra_source=None):
     '''build and install manual pages'''
+    bld.env.SYSTEM_CATALOGS = ''
+    if os.environ.has_key('XML_CATALOG_FILES'):
+	bld.env.SYSTEM_CATALOGS = ' ' + ' '.join('file://' + x for x in os.environ['XML_CATALOG_FILES'].split())
     bld.env.SAMBA_EXPAND_XSL = bld.srcnode.abspath() + '/docs-xml/xslt/expand-sambadoc.xsl'
     bld.env.SAMBA_MAN_XSL = bld.srcnode.abspath() + '/docs-xml/xslt/man.xsl'
     bld.env.SAMBA_CATALOG = bld.srcnode.abspath() + '/bin/default/docs-xml/build/catalog.xml'
-    bld.env.SAMBA_CATALOGS = 'file:///etc/xml/catalog file:///usr/local/share/xml/catalog file://' + bld.env.SAMBA_CATALOG
+    bld.env.SAMBA_CATALOGS = 'file:///etc/xml/catalog file:///usr/local/share/xml/catalog file://' + bld.env.SAMBA_CATALOG + bld.env.SYSTEM_CATALOGS
 
     for m in manpages.split():
         source = m + '.xml'
